### ISSUE TITLE
Audio pop between clips

### ISSUE THREAD

**Issue by tecaspt** [mltframework/shotcut#1392]
Having issues with hearing audio pop/clicks when a new clip starts. It doesn't happen in every single transition, nor does it happen only when I go from file to file, it also happens on cuts within the same video file. The audio sounds something like connecting/disconnecting a mic, it's a very short "pop" as the new clip starts. To reproduce this, just start a new project, add one clip on the timeline, add one after it, remove the gap between them, and export.

I've seen this reported throughout several versions and I do remember this happening to me once in the past, however I either completely forgot what the workaround was, or it's not working anymore. This is happening on versions 22.06.23 and 22.12.21 (I tried updating to see if fixed the issue, it did not), on macOS 13.0.

The issue I'm having is similar to the ones reported below. It seems to pop up here and there on a few versions:
https://forum.shotcut.org/t/audio-popping-sound/18103
https://forum.shotcut.org/t/splitting-audio-adds-pops-clicks/24903/22
https://forum.shotcut.org/t/cant-get-rid-of-audio-popping-sounds/19149
https://forum.shotcut.org/t/audio-crackling-popping/35253
https://forum.shotcut.org/t/audio-crackling-popping-in-preview-exported-video/24521
https://forum.shotcut.org/t/audio-crackling-issue/19285
https://forum.shotcut.org/t/audio-echoes-and-distortion-pops/29577/19
https://forum.shotcut.org/t/slight-blip-pop-when-copying-and-pasting-clips/11182
https://forum.shotcut.org/t/popping-sound-and-visual-glitches-in-exported-videos/15229/16
https://github.com/mltframework/shotcut/issues/314

Opening a new issue here as suggested in the last link posted above

**Comment by bmatherly** [comment]
What is the audio sample rate of your source clips. There is a known issue that sample rate conversion can cause clicks on splits.

**Comment by tecaspt** [comment]
> What is the audio sample rate of your source clips. There is a known issue that sample rate conversion can cause clicks on splits.

Clips are from two separate cameras, one at 44100 and another at 48000

**Comment by bmatherly** [comment]
> Clips are from two separate cameras, one at 44100 and another at 48000

Your pop is probably caused by the sample rate conversion. We have some work-in-progress that is expected to improve/fix this: https://github.com/mltframework/mlt/pull/844

I will ping this thread when we have something available to test.

In the meantime, the workaround is to convert your clips to have 48khz audio. There is an option in the convert dialog to do this, or you can use other tools.

![image](https://user-images.githubusercontent.com/821968/209600222-6210f275-174f-4c5e-bf22-04a2693782a5.png)

**Comment by bmatherly** [comment]
> I have converted the videos that were at 44100 to 48000 and the issue remains.

If that is the case, then your problem must be something else. I only observe the problem when using 44100 clips. The next step would be for you to provide a minimum example project to demonstrate the problem so that a developer can reproduce the problem and investigate.

> I'm wondering if there's a way to convert the audio only?

Not using Shotcut. But other tools like FFMpeg support this.

**Comment by tecaspt** [comment]
> > I have converted the videos that were at 44100 to 48000 and the issue remains.
> 
> If that is the case, then your problem must be something else. I only observe the problem when using 44100 clips. The next step would be for you to provide a minimum example project to demonstrate the problem so that a developer can reproduce the problem and investigate.

I've created a new project and the issue persists, both on original and converted clips. Sending original, converted, project file and export attached: https://drive.google.com/drive/folders/1KS-Iqsfuf4Ft2mDDCXwCwv-h8tyODqzX?usp=share_link

**Comment by tecaspt** [comment]
Seems this is indeed something else as I've tried two other softwares and I'm having similar issues to Shotcut, in varying degrees. This is fairly strange and not something I've experienced before in this or any other camera I've owned. I'll disguise this with audio fade ins/outs while I can't figure out what's up. The odd thing is, this only happens on edited footage, for some reason... So maybe no bugs needing to be fixed here, but perhaps something Shotcut can have tools to assist in avoiding in the future (if possible at all)
